packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactTextShadowNodeUtils.kt

```diff
@@ -0,0 +1,112 @@
+package com.datadog.reactnative.sessionreplay
+
+import android.annotation.SuppressLint
+import android.graphics.Typeface
+import android.util.Log
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ReactShadowNode
+import com.facebook.react.uimanager.UIImplementation
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.views.text.ReactTextShadowNode
+import com.facebook.react.views.text.TextAttributes
+import okhttp3.internal.notify
+import okhttp3.internal.wait
+
+internal class ReactTextShadowNodeUtils(
+    private val reactContext: ReactContext
+) {
+    internal fun getFontSize(view: TextView, shadowNode: ReactTextShadowNode?): Long {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mTextAttributes")?.let {
+            it.isAccessible = true
+            val textAttributes = it.get(shadowNode) as TextAttributes
+            return textAttributes.effectiveFontSize.toLong()
+        }
+
+        return view.textSize.toLong()
+    }
+
+    internal fun getFontFamily(view: TextView, shadowNode: ReactTextShadowNode?): String {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mFontFamily")?.let {
+            it.isAccessible = true
+            return it.get(shadowNode)?.toString()
+                ?: resolveFontFamily(view.typeface)
+        }
+
+        return resolveFontFamily(view.typeface)
+    }
+
+    internal fun getGravity(view: TextView, shadowNode: ReactTextShadowNode?): Int {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mTextAlign")?.let {
+            it.isAccessible = true
+            return it.getInt(shadowNode)
+        }
+
+        return view.gravity
+    }
+
+    internal fun getColor(view: TextView, shadowNode: ReactTextShadowNode?): Int {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mColor")?.let {
+            it.isAccessible = true
+            return it.getInt(shadowNode)
+        }
+
+        return view.currentTextColor
```

Comment:
Question: did you test these fallbacks with real cases? 
If not, are you confident they won't crash the app?
If we're not so sure these are safe, we could return nothing instead, so we avoid crashing the app, that's more important than having support for edge cases.

I think it would be good to add a comment to explain how to test them if we know how to.